dm/syncer/syncer.go

```diff
@@ -3312,6 +3312,9 @@ func (s *Syncer) loadTableStructureFromDump(ctx context.Context) error {
 					zap.Error(err))
 				setFirstErr(err)
 			}
+			// TODO: we should save table checkpoint here, but considering when
```

Comment:
here too
https://github.com/pingcap/tiflow/blob/be428b781aa9cdb67f78db04bc04d9e8670f1b72/dm/syncer/syncer.go#L806

suppose a incremental task:
- upperstream `create table t(c1 int primary key, c2 int)` and insert some rows
- downstream mistake created as `create table t(c1 int primary key)`
- start task
- task fail and table point rollback, and auto resume will use downstream ddl and exec it into schema tracker
- downstream add c2
- will fail now.

Reply:
I'll address comment and add a test about this case tomorrow

Reply:
can't reproduce

```diff
diff --git a/dm/tests/tracker_ignored_ddl/conf/dm-task.yaml b/dm/tests/tracker_ignored_ddl/conf/dm-task.yaml
index a1dc02ebc..cabf94954 100644
--- a/dm/tests/tracker_ignored_ddl/conf/dm-task.yaml
+++ b/dm/tests/tracker_ignored_ddl/conf/dm-task.yaml
@@ -16,7 +16,7 @@ mysql-instances:
     mydumper-config-name: "global"
     loader-config-name: "global"
     syncer-config-name: "global"
-    filter-rules: ["user-filter-1"]
+    filter-rules: ["user-filter-1", "user-filter-2"]
 
 black-white-list: # compatible with deprecated config
   instance:
@@ -28,6 +28,11 @@ filters:                     # filter rules, mysql instance can ref rules in it
     table-pattern: "*"        # pattern of the upstream table name, wildcard characters (*?) are supported
     events: ["alter table"]  # ignore alter table ddl
     action: Ignore
+  user-filter-2:
+    schema-pattern: "tracker_ignored_ddl"
+    table-pattern: "down"
+    events: ["all ddl"]
+    action: Ignore
 
 mydumpers:
   global:
```

```diff
diff --git a/dm/tests/tracker_ignored_ddl/run.sh b/dm/tests/tracker_ignored_ddl/run.sh
index 3fd1d611e..1882dfec5 100644
--- a/dm/tests/tracker_ignored_ddl/run.sh
+++ b/dm/tests/tracker_ignored_ddl/run.sh
@@ -62,6 +62,42 @@ function run() {
        run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
                "query-status test" \
                "\"stage\": \"Running\"" 1
+
+       run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+               "stop-task test" \
+               "\"result\": true" 2
+
+       # to make use of checkpoint like an incremental task
+       test_track_downstream
+}
+
+function test_track_downstream() {
+  echo ">>>>>>>>>>>>>enter test_track_downstream"
+       run_sql_source1 "create table tracker_ignored_ddl.down (c int primary key, c2 int);"
+       run_sql_source1 "insert into tracker_ignored_ddl.down values (1, 1);"
+
+       # downstream created wrong table
+       run_sql_tidb "create table tracker_ignored_ddl.down (c int primary key);"
+
+  run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "start-task $cur/conf/dm-task.yaml"
+
+       run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+               "query-status test" \
+               "Error 1054: Unknown column" 1
+
+       run_sql_tidb "alter table tracker_ignored_ddl.down add column c2 int;"
+       run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+               "resume-task test" \
+               "\"result\": true" 2
+       sleep 3
+       run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+               "query-status test" \
+               "\"stage\": \"Running\"" 1
+
+  read -p 12312123123
+       cleanup_data $TEST_NAME
+       cleanup_process
 }
 
 cleanup_data $TEST_NAME
```

Reply:
https://github.com/pingcap/tiflow/blob/a74c6110c658f199fba88738dc215ec741ef4d91/dm/syncer/checkpoint.go#L935 will cleanup schema-tracker table without checkpoint, so no such issue.